src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Boolean.cs

```diff
@@ -33,6 +37,47 @@ public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
         {
             yield return new[] { TexlStrings.BooleanArg1 };
         }
+
+        public override bool CheckTypes(TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.AssertAllValid(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+            Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
+
+            nodeToCoercedTypeMap = null;
+
+            var isValid = true;
+            var argType = argTypes[0];
+            if (!(DType.Boolean.Accepts(argType) || DType.String.Accepts(argType)))
+            {
+                errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrNumberOrStringExpected);
+                isValid = false;
+            }
+
+            returnType = DType.Boolean;
+            return isValid;
+        }
+
+
+        /// <summary>
+        /// If arg is BoolLit node, no need to make a function call to boolean function. It can just emit arg directly.
+        /// </summary>
+        internal override IR.Nodes.IntermediateNode CreateIRCallNode(PowerFx.Syntax.CallNode node, IRTranslator.IRTranslatorContext context, IRTranslator.IRTranslatorVisitor visitor, ScopeSymbol scope, Features features)
+        {
+            var arg = node.Args.Children[0];
+            if(arg is BoolLitNode)
```

Comment:
>arg is BoolLitNode [](http://example.com/codeflow?start=15&length=18)

We should check type, not BoolLitNode. 

Ie,  `Boolean(x)`  should just be `x`  if x is typed bool. 


A test case to add that I think will catch this:
Boolean(true or true) // true